aiida_firecrest/scheduler.py

```diff
@@ -354,6 +392,52 @@ def kill_job(self, jobid: str) -> bool:
             transport._client.cancel(transport._machine, jobid)
         return True
 
+    def _convert_time(self, string: str) -> int | None:
+        """
+        Note: this function was copied from the Slurm scheduler plugin
+        Convert a string in the format DD-HH:MM:SS to a number of seconds.
```

Comment:
If its copied, can't we just import it? If it is different, can you specify changes?

Reply:
Only because it wasn't an static method in slurm plugin, I couldn't import directly. 
Also we needed to add  "N/A" :
```python
if string == "NOT_SET" or string == "N/A":
     return None
```
But you are right, maybe it's better to make static there and add this string, directly there. The only thing is, this "N/A" is `FirecREST` specific, so doesn't fit in `slurm` scheduler plugin.
What do you think?

Reply:
Have you considered to inherit from the `SlurmScheduler`? From what I understood it is actually a `FirecrestSlurmScheduler`. Then the code would be just
```python
def _convert_time(self, string: str) -> int | None:
    if string == "NOT_SET" or string == "N/A":
         return None
    return super()._convert_time(string)
```

Reply:
Alright, will do this in a separate commit

Reply:
Ok, @agoscinski Just investigated this, which seems more challenging after merged PR [#6043](https://github.com/aiidateam/aiida-core/pull/6043)

`SlurmScheduler` is now inheriting from `BashCliScheduler` which inherits from `Scheduler`.
Ideally, we want `FirecrestScheduler` to inherit only from `Scheduler`. The reason is `BashCliScheduler` as it's name speaks was designed for schedulers that can execute commands, with many abstract methods as such.

If we inherit from `SlurmScheduler` for only one or two tiny functions, then we have to override too many methods as `NotImplemented`. 

To me seems feasible that `FirecrestScheduler` won't inherit from `BashCliScheduler` in a transitive manner.

If you agree I opened the issue https://github.com/aiidateam/aiida-firecrest/issues/42 on this, if needed to be addressed in an separate PR, with possible changes in `aiida-core`